### ISSUE TITLE
ContentTemplateSelector doesn't work in TabControl

### ISSUE THREAD

**Issue by Lapin1916** [MahApps/MahApps.Metro#2649]
## What steps will reproduce this issue?
Hi,MahApps is really awsome!But I got a problem recently that when I used ContentTemplateSelector and Controls.AnimatedSingleRowTabControl.xaml together,the ContentTemplateSelector didn't work.

```xaml
<TabControl Grid.Row="1" Grid.Column="1" ItemsSource="{Binding TabItems}" ContentTemplateSelector="{StaticResource tccts}" >
            <TabControl.ItemTemplate>
                <DataTemplate>
                    <TextBlock Text="{Binding TabName}"/>
                </DataTemplate>
            </TabControl.ItemTemplate>
        </TabControl>`
```

```csharp
public class TabControlContentTemplateSelector : DataTemplateSelector
    {
        public DataTemplate  ShowDataTemplate { get; set; }
        public DataTemplate  HomeTemplate { get; set; }
        public override DataTemplate SelectTemplate(object item, DependencyObject container)
        {
            if (item is MeterDataViewModel)
            {
                return ShowDataTemplate;
            }
            else if (item is HomeViewModel)
            {
                return HomeTemplate;
            }
            else
            {
                return base.SelectTemplate(item, container);
            }
        }
    }
```

```xaml
<lcvm:TabControlContentTemplateSelector x:Key="tccts">
                <lcvm:TabControlContentTemplateSelector.ShowDataTemplate>
                    <DataTemplate>
                        <TextBlock Text="I'm datas."/>
                    </DataTemplate>
                </lcvm:TabControlContentTemplateSelector.ShowDataTemplate>
                <lcvm:TabControlContentTemplateSelector.HomeTemplate>
                    <DataTemplate>
                        <TextBlock Text="Hi." />
                    </DataTemplate>
                </lcvm:TabControlContentTemplateSelector.HomeTemplate>
            </lcvm:TabControlContentTemplateSelector>
```

in MainViewModel‘s constructor:

```csharp
TabItems.Add(new HomeViewModel("test11111111111")); TabItems.Add(new MeterDataViewModel("test5"));
```

When I commented `Controls.AnimatedSingleRowTabControl.xaml` (without singlerow style)，it’s normal. Please help. Thanks.

### Expected outcome

Tabs in single row and ContentTemplateSelector works.

--
### Environment

- MahApps.Metro __v1.2.4__
- Windows __10__
- Visual Studio __2015__
- .NET Framework __4.6__

**Comment by Lapin1916** [comment]
I checked MSDN and found that "If both the ContentTemplateSelector and the ContentTemplate properties are set, then this property is ignored."
MetroAnimatedTabControl.xaml：
` <ContentPresenter x:Name="PART_SelectedContentHost"
                                              UseLayoutRounding="False"
                                              Margin="{TemplateBinding Padding}"
                                              Content="{TemplateBinding SelectedContent}"
                                              ContentSource="SelectedContent"
                                              ContentStringFormat="{TemplateBinding SelectedContentStringFormat}"
                                              ContentTemplate="{TemplateBinding SelectedContentTemplate}"
                                              SnapsToDevicePixels="{TemplateBinding SnapsToDevicePixels}" />`
Is the problem caused by this?

**Comment by punker76** [comment]
@Lapin1916 It's a bug in `MetroAnimatedSingleRowTabControl.xaml` and `Controls.AnimatedSingleRowTabControl.xaml`. I'll fix this.

**Comment by punker76** [comment]
@Lapin1916 Fixed with 9a8fddaa48d019b2757a4ef55979c8ae3b88c20b
